PWGCF/FemtoUniverse/Tasks/femtoUniverseEfficiencyBase.cxx

```diff
@@ -69,14 +71,16 @@ struct femtoUniverseEfficiencyBase {
   Configurable<float> ConfPtHighPart1{"ConfPtHighPart1", 2.5, "Higher limit for Pt for the first particle"};
 
   /// Partition for particle 1
-  Partition<FemtoFullParticles> partsOneMCGen = (aod::femtouniverseparticle::partType == uint8_t(aod::femtouniverseparticle::ParticleType::kMCTruthTrack)) && (ConfNoPDGPartOne || aod::femtouniverseparticle::pidcut == uint32_t(ConfPDGCodePartOne)) &&
-                                                aod::femtouniverseparticle::pt < ConfPtHighPart1 && aod::femtouniverseparticle::pt > ConfPtLowPart1&& nabs(aod::femtouniverseparticle::eta) < ConfEtaMax;
+  Partition<FemtoFullParticles> partsOneMCGen = (aod::femtouniverseparticle::partType == uint8_t(aod::femtouniverseparticle::ParticleType::kMCTruthTrack)) && aod::femtouniverseparticle::pt < ConfPtHighPart1 && aod::femtouniverseparticle::pt > ConfPtLowPart1&& nabs(aod::femtouniverseparticle::eta) < ConfEtaMax;
 
-  Partition<FemtoFullParticles> partsTrackOneMCReco = (aod::femtouniverseparticle::partType == uint8_t(ConfParticleTypePartOne));
```

Comment:
Let's not loose this functionality, please add it ;)

Reply:
I moved check of particle type in doMCRec.. functions. We noticed problems with code functionality when it was in the partition. It works now